<launch>
	<arg name="camera" default="camera"/>
	<arg name="camera_id" default="0" />
	<arg name="image_width" default="640" />
	<arg name="image_height" default="480" />
	<arg name="publish_rate" default="15"/>
	<arg name="machine" default=""/>
	<arg name="info_url" default=""/>

	<node name="$(arg camera)" pkg="simple_webcam" type="camera.py" machine="$(arg machine)" >
		<param name="camera_id" value="$(arg camera_id)"/>
    		<param name="image_width" value="$(arg image_width)"/>
		<param name="image_height" value="$(arg image_height)"/>
	 	<param name="publish_rate" value="$(arg publish_rate)"/>
		<param name="info_url" value="$(arg info_url)"/>
	</node>

	<env name="ROS_NAMESPACE" value="$(arg camera)"/>
	<node name="$(arg camera)_image_proc" pkg="image_proc" type="image_proc" machine="$(arg machine)"/>

    <!-- frames -->
    <arg name="pi/2" value="1.5707963267948966" />
    <arg name="optical_rotate" value="0 0 0 -$(arg pi/2) 0 -$(arg pi/2)" />

    <node pkg="tf" type="static_transform_publisher" name="$(arg camera)_base_link1"
        args="0 -0.045 0 0 0 0  /$(arg camera)_link /$(arg camera)_rgb_frame 100" />

    <node pkg="tf" type="static_transform_publisher" name="$(arg camera)_base_link3"
        args="$(arg optical_rotate) /$(arg camera)_rgb_frame /$(arg camera)_rgb_optical_frame 100" />  

</launch>
